pwndbg/aglib/kernel/nftables.py

```diff
@@ -0,0 +1,629 @@
+from __future__ import annotations
+
+import functools
+from typing import Iterator
+from typing import List
+from typing import Optional
+
+import pwndbg.aglib.memory
+import pwndbg.aglib.symbol
+import pwndbg.aglib.typeinfo
+import pwndbg.dbg
+from pwndbg.aglib.kernel.macros import container_of
+from pwndbg.aglib.kernel.macros import for_each_entry
+
+NFPROTO_INET = 1
+NFPROTO_IPV4 = 2
+NFPROTO_ARP = 3
+NFPROTO_NETDEV = 5
+NFPROTO_BRIDGE = 7
+NFPROTO_IPV6 = 10
+
+nftables_table_family = {
+    "inet": NFPROTO_INET,
+    "ip": NFPROTO_IPV4,
+    "arp": NFPROTO_ARP,
+    "netdev": NFPROTO_NETDEV,
+    "bridge": NFPROTO_BRIDGE,
+    "ip6": NFPROTO_IPV6,
+}
+
+
+def catch_error(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except pwndbg.dbg_mod.Error as e:
+            return f"<unavailable: {e}>"
+        except UnicodeDecodeError as e:
+            return f"<unavailable: {e}>"
+
+    return wrapper
+
+
+class NftFields(object):
+    @catch_error
+    def __getattr__(self, name: str):
+        t = self.__annotations__.get(name)
```

Comment:
I wonder if this should be here or in some base class (if a base class allows to provide getattr for its children?)